File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode *tail = head;
        long cnt = 0;
        while(tail != nullptr){
            cnt++;
            tail = tail -> next;
            if(cnt > 10000)
                return true;
        }
        return false;
    }
};

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode tail = head;
        int cnt = 0;
        while(tail != null){
            cnt++;
            tail = tail.next;
            if(cnt > 10000)
                return true;
        }
        return false;
    }
}